### DISCUSSION TITLE
Previous user still logged in when authenticating with OAuth (Google) and new user is not created.

### DISCUSSION THREAD

**Discussion by g1sbi** [pocketbase/pocketbase#2996] [Q&A] [answered]
I signed up as a user using a standard email/password flow. I then started working on OAuth with Google. The Google I account I used does not share the email of the first account. But when logging the output of the OAuth:
```ts
export async function createOAuthUser() {
  pb.authStore.clear();
  const user = await pb
    .collection("users")
    .authWithOAuth2({ provider: "google" });

  console.log(user)
}
```
I get the previous user. This happens even by changing browsers. Moreover, the new user is not signed up in the database. What am I doing wrong?

**Comment by ganigeorgiev** [comment, marked as answer]
This is most likely because you were already authenticated and the OAuth2 account was linked to your existing user (_we automatically send the existing `pb.authStore` token with the OAuth2 request_).

This is done to allow OAuth2 account linking when the provider doesn't return an email address (_or when it is not verified_).

This type of linking works only if the OAuth2 account isn't already associated to an existing PocketBase user (aka. when you are authenticating for the first time with it).

> This happens even by changing browsers. 

This is because it is already linked. You can find your PocketBase user and "unlink" it from the "Authorized providers" user form panel in the Admin UI:
![panel_screenshot](https://i.imgur.com/PEpSS0T.png)

If you don't want this type of linking, you can call `pb.authStore.clear()` before the `authWithOAuth2()` call.

**Reply by g1sbi** [reply]
That was it. However, after signing up now I get a response with just the token and nothing else.
``` 
Object { meta: {…}, record: {…}, token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb2xsZWN0aW9uSWQiOiJfcGJfdXNlcnNfYXV0aF8iLCJleHAiOjE2OTE1MTkzODksImlkIjoiYmdjeXRjazZ3OXBvNzg0IiwidHlwZSI6ImF1dGhSZWNvcmQifQ.BCJ6GII87V_CYpv2IKbXAbTcaWtfWfyNdPXBX8y2bqY" }
pocketbase.ts:41:17
```

**Reply by ganigeorgiev** [reply]
I'm not sure I follow. The above looks like a valid OAuth2 auth response. Under `meta` are the OAuth2 account data. and `record` contains the PocketBase user. What other fields do you expect?

**Reply by g1sbi** [reply]
Apologies. My dev console wouldn't let me expand the object (which it usually does) so I assumed they were empty.
